lib/metasploit/framework/data_service/proxy/workspace_data_proxy.rb

```diff
@@ -36,8 +36,9 @@ def default_workspace
 
   def workspace
     begin
-      if @current_workspace
-        @current_workspace
+      valid_workspace = find_workspace(@current_workspace.name) if @current_workspace
```

Comment:
This could get expensive if we are asking the db layer for the current workspace often.